Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -13,58 +13,182 @@
+  else
+  {
+    s_image_mask_bounds[index] = value;
+
+    if (!image_mask_cleared)
+    {
+      GLint old_fbo;
+      glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, &old_fbo);
+      glBindFramebuffer(GL_FRAMEBUFFER, image_mask_fbo);
+
+      GLfloat old_clearcolor[4];
+      glGetFloatv(GL_COLOR_CLEAR_VALUE, old_clearcolor);
+      glClearColor(0.f, 0.0f, 0.f, 0.f);
+      glClear(GL_COLOR_BUFFER_BIT);
+      image_mask_updated = false;
+      image_mask_cleared = true;
+
+      glClearColor(old_clearcolor[0], old_clearcolor[1], old_clearcolor[2], old_clearcolor[3]);
+      glBindFramebuffer(GL_FRAMEBUFFER, old_fbo);
+    }
+  }
 }
 
 int BoundingBox::Get(int index)
 {
-  int data = 0;
-  glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
-
-  if (!DriverDetails::HasBug(DriverDetails::BUG_SLOW_GETBUFFERSUBDATA))
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
   {
-    // Using glMapBufferRange to read back the contents of the SSBO is extremely slow
-    // on nVidia drivers. This is more noticeable at higher internal resolutions.
-    // Using glGetBufferSubData instead does not seem to exhibit this slowdown.
-    glGetBufferSubData(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int), &data);
+    int data = 0;
+    glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
+    if (!DriverDetails::HasBug(DriverDetails::BUG_SLOW_GETBUFFERSUBDATA))
+    {
+      // Using glMapBufferRange to read back the contents of the SSBO is extremely slow
+      // on nVidia drivers. This is more noticeable at higher internal resolutions.
+      // Using glGetBufferSubData instead does not seem to exhibit this slowdown.
+      glGetBufferSubData(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int), &data);
+    }
+    else
+    {
+      // Using glMapBufferRange is faster on AMD cards by a measurable margin.
+      void* ptr = glMapBufferRange(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int),
+                                   GL_MAP_READ_BIT);
+      if (ptr)
+      {
+        memcpy(&data, ptr, sizeof(int));
+        glUnmapBuffer(GL_SHADER_STORAGE_BUFFER);
+      }
+    }
+    glBindBuffer(GL_SHADER_STORAGE_BUFFER, 0);
+    return data;
   }
   else
   {
-    // Using glMapBufferRange is faster on AMD cards by a measurable margin.
-    void* ptr = glMapBufferRange(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int),
-                                 GL_MAP_READ_BIT);
-    if (ptr)
+    if (image_mask_updated)
     {
-      memcpy(&data, ptr, sizeof(int));
-      glUnmapBuffer(GL_SHADER_STORAGE_BUFFER);
+      image_mask_updated = false;
+
+      GLint old_texture_2d;
+      glGetIntegerv(GL_TEXTURE_BINDING_2D, &old_texture_2d);
+      glBindTexture(GL_TEXTURE_2D, s_image_mask_texture);
+      glGetTexImage(GL_TEXTURE_2D, 0, GL_RED, GL_UNSIGNED_BYTE, s_image_mask_data);
+      glBindTexture(GL_TEXTURE_2D, old_texture_2d);
+
+      for (int row = 0; row < static_cast<int>(528 * s_yscale); row++)
```

Comment:
That's what I would normally use, but since the bounds are plain `int`s this way avoids a bunch of static casts.